libs/vm/src/module.cpp

```diff
@@ -203,11 +204,16 @@ Module::Module()
   address.CreateMemberFunction("signed_tx", &Address::HasSignedTx);
 
   auto istate = GetClassInterface<IState>();
-  istate.CreateConstuctor<Ptr<String>, TemplateParameter>();
-  istate.CreateConstuctor<Ptr<Address>, TemplateParameter>();
+  istate.CreateConstuctor<Ptr<String>, TemplateParameter1>();
```

Comment:
I think that we should keep the original code here

Reply:
Sure I can do.
I changed it to make the code with intention to make it consistent (with other generic types like Map or PersistentMap).

I would argue we should rather drop the `TemaplateParameter` type alias and use everywhere the indexed type `TemaplateParameter1`.

Reply:
I think I agree with Peter -- drop the TemplateParameter alias and use TemplateParameter1. The actual name of the type inside the compiler and VM, btw, is "TemplateParameter1".

Reply:
...Actually it's "[TempateParameter1]" but the point still stands.